topo_processor/util/files.py

```diff
@@ -0,0 +1,37 @@
+import gzip
+import os
+
+from linz_logger.logger import get_log
+
+
+def is_csv(path: str) -> bool:
+    return path.endswith(".csv")
+
+
+def is_gzip_file(file_path: str) -> bool:
+    with open(file_path, "rb") as file:
+        # gzip magic number == "1f 8b"
+        return file.read(2) == b"\x1f\x8b"
+        # FIXME: check Header
```

Comment:
Something better has to be done

Reply:
the s3 response will include the `Content-Encoding` header which will tell you if the file is gzip encoded. 

otherwise this is the only way to determine if a file on disk is actually gzipd

Reply:
I don't know if testing against `Content-Encoding` from s3 makes sense with the current code as the file is always downloaded?

Reply:
I was more thinking if you download from s3, you can check the content-encoding and then gunzip it, otherwise its is unlikely that a local file will be gzip'd